### PR TITLE
feat(builder): sentry integration

### PR DESCRIPTION
**Pull request by ntindle**
### Background

We want to track errors in the frontend

### Changes 🏗️

Adds sentry and configures a few of its more straightforward settings to meet our goals. The main one I changed was the actions to work

(note 90%+ of this was scaffolded by the sentry wizard 🧙 )

### PR Quality Scorecard ✨

- [x] Have you used the PR description template? &ensp; `+2 pts`
- [ ] Is your pull request atomic, focusing on a single change? &ensp; `+5 pts`
- [ ] Have you linked the GitHub issue(s) that this PR addresses? &ensp; `+5 pts`
- [ ] Have you documented your changes clearly and comprehensively? &ensp; `+5 pts`
- [ ] Have you changed or added a feature? &ensp; `-4 pts`
  - [ ] Have you added/updated corresponding documentation? &ensp; `+4 pts`
  - [ ] Have you added/updated corresponding integration tests? &ensp; `+5 pts`
- [ ] Have you changed the behavior of AutoGPT? &ensp; `-5 pts`
  - [ ] Have you also run `agbenchmark` to verify that these changes do not regress performance? &ensp; `+10 pts`

### CODE REVIEW

**Comment by ntindle** [comment]
> DSNs are safe to keep public because they only allow submission of new events and related event data; they do not allow read access to any information.

https://docs.sentry.io/concepts/key-terms/dsn-explainer/#dsn-utilization

**Comment by Swiftyos** [comment]
What does sentry offer that we are not currently getting with google analytics?

**Comment by ntindle** [comment]
Backend errors?

**Comment by ntindle** [comment]
Traces through our other services

**Review by majdyz — APPROVED** [review]

File: rnd/autogpt_builder/src/components/admin/marketplace/AdminFeaturedAgentsControl.tsx [hunk 1]
```diff
@@ -55,9 +56,15 @@ export default async function AdminFeaturedAgentsControl({
             component: <Button>Remove</Button>,
             action: async (rows) => {
               "use server";
-              const all = rows.map((row) => removeFeaturedAgent(row.id));
-              await Promise.all(all);
-              revalidatePath("/marketplace");
+              return await Sentry.withServerActionInstrumentation(
```
Comment (majdyz) [inline rnd/autogpt_builder/src/components/admin/marketplace/AdminFeaturedAgentsControl.tsx:59]: LGTM, but I think this is definitely worth a follow-up:
The changes wrapping sentry here can be a wrapper function.

Comment (ntindle) [reply rnd/autogpt_builder/src/components/admin/marketplace/AdminFeaturedAgentsControl.tsx:59]: 100% agree

**Pull request merged**